sklearn/utils/validation.py

```diff
@@ -1273,7 +1273,7 @@ def _check_psd_eigenvalues(lambdas, enable_warnings=False):
     return lambdas
 
 
-def _check_sample_weight(sample_weight, X, dtype=None):
+def _check_sample_weight(sample_weight, X, dtype=None, copy=False):
```

Comment:
Wouldn't it be safer to always copy and not have this param ? It's not the first time we see this issue.

What do you think @glemaitre @rth ?

Reply:
I would say that it would be best if we can avoid copies. However, it might be safer to have `copy=True` by default (not sure what it would involved thought).

Reply:
Also +1 for copy=False by default. The function is "check sample weight", nothing says it should make copies. So if someones makes inplace modifications of `sample_weight` IMO it should be up to them to check that a copy was made previously.